number_search.py

```diff
@@ -4,24 +4,38 @@
 import re
 
 def count_numbers(text):
-    """
-        counts whole number and decimals i.e -ve and +ve whole numbers decimals
-        TODO update doc strings
-    """
-    pass #TODO update the fucntion to pass test
+    pattern = r"(?:-?\\frac{\d+}{\d+}|-?\d+\\tinyfrac{\d+}{\d+}|[A-Z]+[0-9]+|[a-z][0-9]+|-?\d+\.? ?,?\d+|-?\d+)"
 
+    with open(text, "r") as f:
+        text = f.read()
+        regex = re.compile(char)
+        matched = regex.findall(text)
+        searched_matches = []
+        for i in range(len(matched)):
+            if re.compile(r"(?:^[-]?\d+\.? ?,?\d+$|^[-]?\d+$)").search(matched[i]):
```

Comment:
good job
this pattern is almost perfect but try to include comma and fullstop...